docs/dev/README.md

```diff
@@ -0,0 +1,218 @@
+# Development guide for aws-crt-python
+
+This guide is for contributors to aws-crt-python's source code.
+Familiarity (but not necessarily expertise) with Python and C is assumed.
+
+### Table of Contents
+
+*   [Git](#git)
+*   [Set up a Virtual Environment](#set-up-a-virtual-environment)
+*   [Install](#install)
+*   [Run Tests](#run-tests)
+    *   [Environment Variables for Tests](#environment-variables-for-tests)
+*   [Using an IDE](#using-an-ide)
+    *   [Visual Studio Code](#using-visual-studio-code-vscode)
+        *   [Debugging Python](#debugging-python-with-vscode)
+        *   [Debugging C](#debugging-c-with-vscode)
+
+## Git
+
+Clone to a development folder:
+```sh
+$ git clone git@github.com:awslabs/aws-crt-python.git
+$ cd aws-crt-python
+$ git submodule update --init
```

Comment:
Trivial - I don't think it's necessary to use --recursive unless we've got nested submodules but it doesn't hurt to put it there in case some changes in the future to how we modularize involves nesting of submodules. It'd be kinda like futureproofing this guide but then again, maybe we never will. Figured I'd mention it because when dengke first showed me how to pull in submodules he added --recursive and I've just been using it since =P

Reply:
you only need the --recursive in one repo: aws-iot-device-sdk-cpp-v2

you don't need it in any of the aws-crt-xyz libraries. I'm going to keep this simple